Mathlib/AlgebraicGeometry/Cover/Over.lean

```diff
@@ -0,0 +1,209 @@
+definitionally equal to `AlgebraicGeometry.Scheme.Cover.pullbackCover`, as here we take
+the pullback in `Over S`, whose underlying schemes is only isomorphic but not equal to the
+pullback in `Scheme`. -/
+@[simps]
+def Cover.pullbackCoverOver : W.Cover P where
+  J := 𝒰.J
+  obj x := (pullback (f.asOver S) ((𝒰.map x).asOver S)).left
+  map x := (pullback.fst (f.asOver S) ((𝒰.map x).asOver S)).left
+  f x := 𝒰.f (f.base x)
+  covers x := by
+    obtain ⟨y, hy⟩ := 𝒰.covers (f.base x)
+    obtain ⟨o, ho⟩ := IsJointlySurjectivePreserving.exists_preimage_fst_triplet_of_prop
+      (𝒰.map_prop _) x y hy.symm
+    use (PreservesPullback.iso (Over.forget S) (f.asOver S) ((𝒰.map _).asOver S)).inv.base o
+    simp only [Over.forget_obj, Over.forget_map, OverClass.asOverHom_left]
+    simp_rw [← Scheme.comp_base_apply, ← Over.forget_map, PreservesPullback.iso_inv_fst]
+    simpa
+  map_prop j := by
+    dsimp only
+    rw [← Over.forget_map, ← PreservesPullback.iso_hom_fst, P.cancel_left_of_respectsIso]
+    exact P.pullback_fst _ _ (𝒰.map_prop j)
+
+instance (j : 𝒰.J) : ((𝒰.pullbackCoverOver S f).obj j).Over S where
+  hom := (pullback (f.asOver S) ((𝒰.map j).asOver S)).hom
+
+instance : (𝒰.pullbackCoverOver S f).Over S where
+  isOver_map j := { comp_over := by exact Over.w (pullback.fst (f.asOver S) ((𝒰.map j).asOver S)) }
+
+/-- A variant of `AlgebraicGeometry.Scheme.Cover.pullbackCoverOver` with the arguments in the
+fiber products flipped. -/
+@[simps]
+def Cover.pullbackCoverOver' : W.Cover P where
+  J := 𝒰.J
+  obj x := (pullback ((𝒰.map x).asOver S) (f.asOver S)).left
+  map x := (pullback.snd ((𝒰.map x).asOver S) (f.asOver S)).left
+  f x := 𝒰.f (f.base x)
+  covers x := by
+    obtain ⟨y, hy⟩ := 𝒰.covers (f.base x)
+    obtain ⟨o, ho⟩ := IsJointlySurjectivePreserving.exists_preimage_snd_triplet_of_prop
+      (𝒰.map_prop _) y x hy
+    use (PreservesPullback.iso (Over.forget S) ((𝒰.map _).asOver S) (f.asOver S)).inv.base o
+    simp only [Over.forget_obj, Over.forget_map, OverClass.asOverHom_left]
+    simp_rw [← Scheme.comp_base_apply, ← Over.forget_map, PreservesPullback.iso_inv_snd]
+    simpa
+  map_prop j := by
+    dsimp only
+    rw [← Over.forget_map, ← PreservesPullback.iso_hom_snd, P.cancel_left_of_respectsIso]
+    exact P.pullback_snd _ _ (𝒰.map_prop j)
+
+instance (j : 𝒰.J) : ((𝒰.pullbackCoverOver' S f).obj j).Over S where
+  hom := (pullback ((𝒰.map j).asOver S) (f.asOver S)).hom
+
+instance : (𝒰.pullbackCoverOver' S f).Over S where
+  isOver_map j := { comp_over := by exact Over.w (pullback.snd ((𝒰.map j).asOver S) (f.asOver S)) }
+
+variable {Q : MorphismProperty Scheme.{u}} [Q.HasOfPostcompProperty Q]
+  [Q.IsStableUnderBaseChange] [Q.IsStableUnderComposition]
+
+variable (hX : Q (X ↘ S)) (hW : Q (W ↘ S)) (hQ : ∀ j, Q (𝒰.obj j ↘ S))
+
+/-- A variant of `AlgebraicGeometry.Scheme.Cover.pullbackCoverOverProp` with the arguments in the
+fiber products flipped. -/
```

Comment:
It seems to me this docstring does not correspond to the actual content. It should be expanded similarly as `Cover.pullbackCoverOver`.